docs/semanticdb/specification.md

```diff
@@ -3573,30 +3623,55 @@ modelled in [Scala symbols](#scala-symbol).
 
 <a name="java-annotation"></a>
 
-### Annotation
+### AnnotationTree
 
 ```protobuf
-message Annotation {
+message AnnotationTree {
   Type tpe = 1;
+  repeated Tree parameters = 2;
 }
 ```
 
-In Java, [Annotation](#annotation) represents `access_flags` in the JVMS `class`
-file format [\[92\]][92] but not the actual annotations [\[93\]][93]. We may
-improve on this in the future.
+In Java, [AnnotationTree](#annotation-tree) represents a typed syntax tree of the JLS annotation construct [\[93\]][93].
 
 <table>
   <tr>
     <td><b>Value</b></td>
     <td><b>Explanation</b></td>
   </tr>
   <tr>
-    <td><code>Annotation(TypeRef(None, &lt;scala/annotation/strictfp&gt;, List()))</code></td>
-    <td> Declared <code>strictfp</code>; floating-point mode is FP-strict e.g. <code>strictfp class MyClass</code>.</td>
+    <td><code>AnnotationTree(TypeRef(None, &lt;NoArgs#&gt;, List()), None)</code></td>
+    <td><code>@NoArgs</code></td>
   </tr>
   <tr>
-    <td>Not supported</td>
-    <td>JLS annotations <a href="https://docs.oracle.com/javase/specs/jls/se8/html/jls-9.html#jls-9.7">[93]</a></td>
+    <td>
+      <code>
+        AnnotationTree(TypeRef(None, &lt;StringArg#&gt;, List()),
+        List(AssignTree(IdTree(&lt;StringArg#value().&gt;),
+          LiteralTree(StringConstant("arg")))))
+      </code>
+    </td>
+    <td><code>@StringArgs("arg")</code></td>
+  </tr>
+  <tr>
+    <td>
+      <code>
+        AnnotationTree(TypeRef(None, &lt;ArrayArg#&gt;, List()),
+        List(AssignTree(IdTree(&lt;ArrayArg#value().&gt;),
+          ApplyTree(IdTree(&lt;scala/Array#&gt;), List())))
+      </code>
+    </td>
+    <td><code>@ArrayArg({})</code></code></td>
+  </tr>
+  <tr>
+    <td>
+      <code>
+        AnnotationTree(TypeRef(None, &lt;EnumConstArg#&gt;, List()),
+        List(AssignTree(IdTree(&lt;EnumConstArg#value().&gt;),
+          SelectTree(IdTree(&lt;X#&gt;), IdTree(&lt;X#CONST.&gt;))))
+      </code>
+    </td>
+    <td><code>@EnumConstArg(X.CONST)</code></td>
```

Comment:
Can we add one example for `AssignTree` and one example for nested `AnnotationTree`